plugins/unicodelib/unicodelib.cpp

```diff
@@ -1695,3 +1707,30 @@ UNICODELIB_API bool UNICODELIB_CALL ulUnicodeLocaleEndsWith(unsigned srcLen, UCh
     }
     return endsWith(pro, suf);
 }
+
+UNICODELIB_API void UNICODELIB_CALL ulUnicodeLocaleRemoveSuffix(unsigned & tgtLen, UChar * & tgt, unsigned srcLen, UChar const * src, unsigned suffLen, UChar const * suff, unsigned formLen, char const * form)
+{
+    UnicodeString pro(src, srcLen);
+    UnicodeString suf(suff, suffLen);
+    suf.trim();
+    if (formLen == 3 || formLen == 4)
+    {
+        normalizationFormCheck(pro, form);
+        normalizationFormCheck(suf, form);
+    }
+    if (endsWith(pro, suf))
+    {
+        removeSuffix(pro, suf);
+    }
+    if (pro.length()>0)
+    {
+        tgtLen = pro.length();
+        tgt = (UChar *)CTXMALLOC(parentCtx, tgtLen * 2);
+        pro.extract(0, tgtLen, tgt);
+    }
+    else
+    {
+        tgtLen = 0;
+        tgt = 0;
+    }
+}
```

Comment:
trivial: It can cause merge problems if there is not final newline on the end of the file.